### PR TITLE
Mysql store

### PR DESCRIPTION
**Pull request by printfcoder**
go micro store for mysql

### CODE REVIEW

**Comment by asim** [comment]
Adding @vtolstov to review

File: store/mysql/mysql.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	}
+	record.Expiry = time.Until(cachedTime)
+	records = append(records, record)
+
+	return records, nil
+}
+
+// Write records
+func (s *sqlStore) Write(r *store.Record) error {
+	q, err := s.db.Prepare(fmt.Sprintf("INSERT INTO %s.%s (`key`, value, expiry) VALUES(?, ?, ?) ON DUPLICATE KEY UPDATE `value`= ?, `expiry` = ?", s.database, s.table))
```
Comment (vtolstov) [inline store/mysql/mysql.go]: also best to prepare query in NewStore

File: store/mysql/mysql.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	DefaultPrefix    = "micro"
+)
+
+type sqlStore struct {
+	db *sql.DB
+
+	database string
+	table    string
+
+	options store.Options
```
[the remark this answers is not in the document]
Comment (printfcoder) [reply store/mysql/mysql.go]: Cool. I will do this next version and abstract it in the store interface

**Comment by printfcoder** [comment]
@vtolstov i updated just now. plz check it again. thanks.

File: store/mysql/mysql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vtolstov) [reply store/mysql/mysql.go]: do you plan to add this before merge or after? (so question is - i need wait for timeout or merge now?)

**Pull request merged**